--- internal/app/machined/pkg/controllers/kubespan/peer_spec.go.orig	2026-03-04 05:49:49.157666239 +0000
+++ internal/app/machined/pkg/controllers/kubespan/peer_spec.go	2026-03-13 15:09:13.672876177 +0000
@@ -46,9 +46,9 @@
 			Kind:      controller.InputWeak,
 		},
 		{
-			Namespace: cluster.NamespaceName,
-			Type:      cluster.IdentityType,
-			ID:        optional.Some(cluster.LocalIdentity),
+			Namespace: kubespan.NamespaceName,
+			Type:      kubespan.IdentityType,
+			ID:        optional.Some(kubespan.LocalIdentity),
 			Kind:      controller.InputWeak,
 		},
 	}
@@ -80,7 +80,7 @@
 			return fmt.Errorf("error getting kubespan configuration: %w", err)
 		}

-		localIdentity, err := safe.ReaderGetByID[*cluster.Identity](ctx, r, cluster.LocalIdentity)
+		localIdentity, err := safe.ReaderGetByID[*kubespan.Identity](ctx, r, kubespan.LocalIdentity)
 		if err != nil && !state.IsNotFoundError(err) {
 			return fmt.Errorf("error getting first MAC address: %w", err)
 		}
@@ -93,20 +93,16 @@
 		touchedIDs := map[resource.ID]struct{}{}

 		if cfg != nil && localIdentity != nil && cfg.TypedSpec().Enabled {
-			localAffiliateID := localIdentity.TypedSpec().NodeID
+			localPublicKey := localIdentity.TypedSpec().PublicKey

 			peerIPSets := make(map[string]*netipx.IPSet, affiliates.Len())

 		affiliateLoop:
 			for affiliate := range affiliates.All() {
-				if affiliate.Metadata().ID() == localAffiliateID {
-					// skip local affiliate, it's not a peer
-					continue
-				}

 				spec := affiliate.TypedSpec()

-				if spec.KubeSpan.PublicKey == "" {
-					// no kubespan information, skip it
+				if spec.KubeSpan.PublicKey == "" || spec.KubeSpan.PublicKey == localPublicKey {
+					// no kubespan information or self, skip it
 					continue
 				}
